unit_tests/test_redis_client_base.py

```diff
@@ -0,0 +1,179 @@
+import json
+import uuid
+import pytest
+from redis import StrictRedis
+import d3a_api_client
+from d3a_api_client.redis_client_base import RedisClientBase, RedisAPIException
+from unittest.mock import MagicMock
+
+area_id = str(uuid.uuid4())
+transactionID = str(uuid.uuid4())
+d3a_api_client.redis_client_base.StrictRedis = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.publish = MagicMock(spec=StrictRedis)
+d3a_api_client.redis_client_base.StrictRedis.pubsub.psubscribe = MagicMock()
```

Comment:
From what I understand, you're mocking these functions because you don't want to execute the real calls while you run the individual unit tests. This is the correct approach (IMO) but it could be improved. Overriding these functions with mocks at the module level is not optimal and could lead to undesired side-effects.

A common way to do this is to use unittest's [`mock.patch()`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.patch) decorator.
You can see an example on the d3a repo: https://github.com/gridsingularity/d3a/blob/3812a5f9574943b630b05999b32022bf784b631b/tests/test_strategy_home_meter.py#L88

Reply:
Yes, Sure I will change it